### DISCUSSION TITLE
Can `.items()` be used with std algorithms like `transform`

### DISCUSSION THREAD

**Discussion by av4625** [nlohmann/json#4398] [Q&A] [answered]
Hi I was saving keys in a vector like this:
```C++
std::vector<std::string> tracks;

const nlohmann::json config(nlohmann::json::parse(f));

for (const auto& track : config.items())
{
   tracks.emplace_back(track.key());
}
```

That was fine but then cppcheck was suggesting to use `std::transform`. I tried the following but get a segfault on the `.key()` call in the lambda:
```C++
std::vector<std::string> tracks;

const nlohmann::json config(nlohmann::json::parse(f));

std::transform(
    config.items().begin(),
    config.items().end(),
    tracks.begin(),
    [](const auto& track){return track.key();});
```

Is doing something like this possible?

I found an issue about using std algorithms here: https://github.com/nlohmann/json/issues/1045
It suggests a fix was done, but maybe not exactly the same issue.

**Comment by gregmarr** [comment, marked as answer]
Are you missing code in the sample?  You need to either set the size of `tracks` to be the same as the number of entries in `config` or use the `back_inserter` wrapper to resize the vector as it goes.
https://godbolt.org/z/xnTMPdTKr

**Reply by av4625** [reply]
Ahhhh, I'm an idiot and `back_inserter` is what I need, thank you for pointing it out!
